### DISCUSSION TITLE
How to compose iPhone/iPad screenshot into bezel

### DISCUSSION THREAD

**Discussion by ogezoos** [ImageMagick/ImageMagick#7730] [Help] [answered]
### ImageMagick version

7.1.1-38

### Operating system, version and so on

macOS 15.0.1

### Description

I am trying to compose an iPhone/iPad screenshot into the bezel. 
here is the result of this command:
magick screenshot.png bezel.png -gravity center -composite output.png
![output](https://github.com/user-attachments/assets/b84218b8-70ba-4673-8135-ffec69b41e0a)
and here is the result of this command in another order
magick bezel.png screenshot.png -gravity center -composite output.png
![output](https://github.com/user-attachments/assets/25d24f1c-02cc-43e7-893d-168df18f29b5)
What I should do to put the screen behind the bezels but not cut out parts of the bezels to the size of the screen?

**Comment by fmw42** [comment]
Please post your two input images separately and your desired result.

**Reply by ogezoos** [reply]
Okay so here is real example. 
Bezel
![bezel](https://github.com/user-attachments/assets/6cb9af51-1061-4087-a474-95cfa963490b)
Screenshot
![screenshot](https://github.com/user-attachments/assets/51bf5cfb-a2d4-4aed-af41-1f7f4a59b6c1)
Desired result
![result](https://github.com/user-attachments/assets/8b6d4352-b6e2-4565-b3c8-c64f92b95aeb)

**Comment by fmw42** [comment, marked as answer]
To solve this you need to make a mask for the inside and then use that mask to control the composite of the filler over the frame.  To make the mask, I flood fill the center of the frame with white and then make everything else black and turn off the alpha channel from the -draw.

```
magick frame.png filler.png \
\( -clone 0 -fill white -draw "color %[fx:w/2],%[fx:h/2] floodfill" -fill black +opaque white -alpha off \) \
-gravity center -compose over -composite result.png

```
![result](https://github.com/user-attachments/assets/b264d4b1-c719-4b9a-83ff-69d86fbb4d99)

**Reply by ogezoos** [reply]
Thanks a lot! One last question: Is there a way to smooth out the edges inside the bezel? I can see there are some imperfections.

**Reply by fmw42** [reply]
blur the mask a little.  try

```
magick frame.png filler.png \
\( -clone 0 -fill white -draw "color %[fx:w/2],%[fx:h/2] floodfill" -fill black +opaque white -alpha off -blur 0x2 \) \
-gravity center -compose over -composite result2.png

```

![result2](https://github.com/user-attachments/assets/ed59a632-c4c0-4d49-8157-a1f01841b52d)


Is that better?  If not, please identify where the imperfections are located.

**Reply by ogezoos** [reply]
Yes, that's it, again thanks a lot!
